srcalc/src/main/java/gov/va/med/srcalc/web/controller/EnterVariablesController.java

```diff
@@ -46,8 +49,19 @@ public VariableEntry constructVariableEntry(
         // Get the CalculationWorkflow from the session.
         final CalculationWorkflow workflow =
                 CalculationController.getWorkflowFromSession(session);
-        return VariableEntry.withRetrievedValues(workflow.getCalculation().getVariables(), 
+        final VariableEntry initialValues = VariableEntry.withRetrievedValues(workflow.getCalculation().getVariables(), 
         		workflow.getCalculation().getPatient());
+        // In the case of using the "Return to Input Form" button, add the values already
+        // in the calculation to the variable entry object to maintain the previously calculated
+        // values on the entry page.
+        final DynamicValueVisitor visitor = new DynamicValueVisitor(initialValues);
```

Comment:
Please add a test to the Controller's IT to verify population of previous values.